sphinx_gallery/py_source_parser.py

```diff
@@ -161,7 +161,7 @@ def split_code_and_text_blocks(source_file):
     file_conf = extract_file_config(rest_of_content)
 
     pattern = re.compile(
-        r'(?P<header_line>^#{20,}.*)\s(?P<text_content>(?:^#.*\s)*)',
+        r'(?P<header_line>^#{20,}.*|^#%%.*|^# %%.*)\s(?P<text_content>(?:^#.*\s)*)',
```

Comment:
```suggestion
        r'(?P<header_line>^#{20,}.*^# ?%%.*)\s(?P<text_content>(?:^#.*\s)*)',
```
I think this should work and is more compact

Reply:
Thanks, don't know why I didn't think of that!